### PR TITLE
Refactored the TTS subsystem to more properly handle message garbling. Added a volume preference for TTS.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request
TTS subsystem now uses datums to handle requests. Also removed the DM-side caching since it wasn't really all that useful compared to the complexity it added to the entire system.

Messages will no longer play at the same time the request was received. This is so that the TTS audio files don't play out of order.
Also adds a preference for players to adjust the volume of TTS sounds.
Additionally adds a debug verb to re-establish connection to the TTS server.

Removes double processing when calculating the listeners for TTS messages as well, which may fix some issues where you'd be able to hear a message that you weren't supposed to hear.
Fixed being able to hear messages whilst unconscious by refactoring the Hear proc so that it returns a boolean value for whether the message was understood or not.

## Why It's Good For The Game
Fixes message garbling. Also gives players a way to control the volume of TTS messages for themselves.

## Changelog
:cl: Watermelon914, Iamgoofball
fix: Fixed message garbling. TTS messages won't play in the wrong order anymore.
qol: Added a way for players to change the volume of TTS.
/:cl:

### CODE REVIEW

**Comment by Watermelon914** [comment]
Needs python server updates before this can be TM'd

File: tools/tts/tts/tts.py [hunk 1]
```diff
@@ -4,18 +4,23 @@
...
 import json
 import gc
+import random
 from flask import Flask, request, send_file
+from pydub import AudioSegment
+from pydub.silence import split_on_silence
 
 tts = TTS("tts_models/en/vctk/vits", progress_bar=False, gpu=False)
-
+letters_to_use = "abcdefghijklmnopqrstuvwxyz1234567890"
```
Comment (MrStonedOne) [inline tools/tts/tts/tts.py:13]:
```suggestion
letters_to_use = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
```

File: tools/tts/tts/tts.py [hunk 2]
```diff
@@ -37,6 +42,54 @@ def text_to_speech():
 	request_count += 1
 	return result
 
+@app.route("/generate-tts-blips")
+def text_to_speech_blips():
+	global request_count
+	text = request.json.get("text", "")
```
Comment (MrStonedOne) [inline tools/tts/tts/tts.py:48]:
```suggestion
	text = request.json.get("text", "").upper()
```

File: tools/tts/tts/tts.py [hunk 3]
```diff
@@ -37,6 +42,54 @@ def text_to_speech():
...
+		voice = voice_name_mapping_reversed[voice]
+
+	result = None
+	with io.BytesIO() as data_bytes:
+		with torch.no_grad():
+			result_sound = None
+			if not os.path.exists('samples/' + voice):
+				os.makedirs('samples/' + voice, exist_ok=True)
+				for i, value in enumerate(letters_to_use):
+					tts.tts_to_file(text=value.upper() + ".", speaker=voice, file_path="samples/" + voice + "/" + value + ".wav")
```
Comment (MrStonedOne) [inline tools/tts/tts/tts.py:60]:
```suggestion
					tts.tts_to_file(text=value + ".", speaker=voice, file_path="samples/" + voice + "/" + value + ".wav")
```

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
Goof made this not work on linux because he forgot case matters in filenames on linux.

**Pull request merged**